test/histogram.test.js

```diff
@@ -0,0 +1,115 @@
+const { expect } = require('chai');
+const HistogramTrace = require('../data/histogramTrace');
+const Histogram = require('../charts/histogram');
+
+describe('HistogramChart Tests', () => {
+  it('expects a HistogramTrace object', () => {
+    const x1 = [];
+    const x2 = [];
+    const y1 = [];
+    const y2 = [];
+    for (let i = 1; i < 500; i += 1) {
+      const k = Math.random();
+      x1.push(k * 5);
+      x2.push(k * 10);
+      y1.push(k);
+      y2.push(k * 2);
+    }
+
+    const trace = {
+      x: x1,
+      y: y1,
+      name: 'Trace1',
+    };
+
+    const histogramTrace = new HistogramTrace(trace);
+    expect(histogramTrace.type).to.equal('histogram');
+    expect(histogramTrace.x).to.have.lengthOf(x1.length);
+    expect(histogramTrace.y).to.have.lengthOf(y1.length);
+    expect(histogramTrace.name).to.equal('Trace1');
+  });
+
+  it('expects a Histogram Chart object', () => {
+    const x1 = [];
+    const x2 = [];
+    const y1 = [];
+    const y2 = [];
+    for (let i = 1; i < 500; i += 1) {
+      const k = Math.random();
+      x1.push(k * 5);
+      x2.push(k * 10);
+      y1.push(k);
+      y2.push(k * 2);
+    }
+
+    const histogramTrace1 = {
+      x: x1,
+      y: y1,
+      name: 'Trace1',
+      autobinx: true,
+      marker: {
+        color: undefined,
```

Comment:
Is there a reason for setting all those properties to `undefined`? Because, I don't see a difference:
```
var obj = {x: undefined};
console.log(obj.x); // --> undefined
console.log(obj.y); // --> undefined
```

Reply:
The main reason is we can see which fields are used for that specific chart. Even those fields are not assigned, developers can check what type of changes can be made in the chart objects.